WordPress/gutenberg
# Replace block variation buttons with ToggleGroupControl

## What?
The buttons to transform blocks to different variantions should be replaced with the new icon button variant of ToggleGroupControl.

## Why?
Because the ToggleGroupControl  is better suited for this. This will improve consistency, reduce code duplication, and add visual/semantic indications that the options are mutually exclusive.

## How?
I replaced the buttons with `ToggleGroupControl`/`ToggleGroupControlOptionIcon`

## Testing Instructions
1. Create a new post or page
2. Add a Group, Row or Column block
3. Verify that the block variations picker still works.

## Screenshots or screencast 
| Before | After |
|-------|-------|
| <img width="278" alt="CleanShot 2022-11-09 at 12 00 15@2x" src="https://user-images.githubusercontent.com/528287/200918978-8cb653f2-58ad-4284-befb-3944638cab5c.png"> | <img width="274" alt="CleanShot 2022-11-09 at 11 57 36@2x" src="https://user-images.githubusercontent.com/528287/200919020-7c4b9320-f255-4ec7-9631-62c611893bf7.png"> |
